File: BinarySearch.java
public class BinarySearch {
     int Search(int[] arr,int tar){
        int left=0;
        int right = arr.length-1;
        while(left<=right){
            int middle = (left+right)/2;
            if(arr[middle] == tar){
                return middle;
            }
            if(tar < arr[middle]){
                right = middle-1;
            }
            else{
                left = middle+1;
            }
        }
        return -1;
    }
    public static void main(String args[]){

        int[] numbers = {10,20,30,40,50,60,70};
        int target = 70;
        BinarySearch b = new BinarySearch();
        int result = b.Search(numbers,target);
        System.out.println(result);
    }
}

File: python/binarySearch.py
def binarySearch(numbers, target):
    left = 0;
    right = len(numbers)-1;
    #middle = left + (right-left) // 2
    while left <= right:
        middle = (left + right) // 2;
        if(numbers[middle] == target):
            return middle;
        elif target < numbers[middle]:
            right = middle+1;
        else:
            left = middle-1;
            
    return -1;

numbers = [10,20,30,35,40,250,1000]

result = binarySearch(numbers, 250);
print('result',result);